src/Data/Tree/Binary/Zipper.agda

```diff
@@ -0,0 +1,108 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Zippers for Binary Trees
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Tree.Binary.Zipper where
+
+open import Level using (Level; _⊔_)
+open import Data.Tree.Binary as BT using (Tree; node; leaf)
+open import Data.List as List using (List; []; _∷_; sum; _++_; [_])
+open import Data.Maybe using (Maybe; nothing; just)
+open import Data.Nat.Base using (ℕ; suc; _+_)
+open import Function
+
+private
+  variable
+    a b c d : Level
+    A : Set a
+    B : Set b
+    C : Set c
+    D : Set d
+
+data Crumb (A : Set a) (B : Set b) : Set (a ⊔ b) where
+  leftBranch : A → Tree A B → Crumb A B
+  rightBranch : A → Tree A B → Crumb A B
+
+record Zipper (A : Set a) (B : Set b) : Set (a ⊔ b) where
+  constructor mkZipper
+  field
+    context : List (Crumb A B)
+    focus : Tree A B
+
+open Zipper public
+
+-- Fundamental operations of a Zipper: Moving around
+------------------------------------------------------------------------
```

Comment:
We usually put the long `-----------` above the section title, rather than below.

Reply:
I think I copied this from the list zipper file, though I am happy to change it

Reply:
Ah okay, that obviously slipped past me last time! I'd never condone something so obviously wrong as putting the dashes below rather than above :smile: If you wouldn't mind changing it that'd be great, thanks.